pkg/engine/internal/types/grouping.go

```diff
@@ -0,0 +1,30 @@
+package types
+
+import "fmt"
+
+// GroupingMode represents the grouping by/without label(s) for vector aggregators and range vector aggregators.
+type GroupingMode int
+
+const (
+	GroupingModeInvalid         GroupingMode = iota
+	GroupingModeByEmptySet                   // Grouping by empty label set: <operation> by () (<expr>)
```

Comment:
do we need 4 of these? `GroupingModeByEmptySet` is the same as `GroupingModeByLabelSet` without any groupings right 

This can also be a flag. `without=(true|false)`